### PR TITLE
[CSL-2087] More logs in `getHistory`, remove huge ones

### PR DESCRIPTION
**Pull request by adinapoli-iohk**
This is cherry pick of https://github.com/input-output-hk/cardano-sl/commit/17a68ec565ba5cdd0d7e8e662bd52294ed9fe5e7

See #2278 for the full rationale.

### CODE REVIEW

File: wallet/src/Pos/Wallet/Web/Methods/History.hs [hunk 1]
```diff
@@ -55,20 +58,25 @@ getFullWalletHistory cWalId = do
...
+
     -- TODO when we introduce some mechanism to react on new tx in mempool,
     -- we will set timestamp tx as current time and remove call of @addHistoryTxs@
     -- We call @addHistoryTxs@ only for mempool transactions because for
     -- transactions from block and resubmitting timestamp is already known.
     addHistoryTxs cWalId localHistory
-    cHistory <- forM fullHistory (constructCTx cWalId walAddrs diff)
+    logDebug "getFullWalletHistory: invoked addHistoryTxs"
+
+    cHistory <- forM fullHistory (constructCTx cWalId cAddrsSet diff)
```
Comment (dcoutts) [inline wallet/src/Pos/Wallet/Web/Methods/History.hs]: If we want to force this map, then we should do it here. See comment below.

File: wallet/src/Pos/Wallet/Web/Methods/History.hs [hunk 2]
```diff
@@ -94,7 +102,9 @@ getHistory cWalId accIds mAddrId = do
             | addr `S.member` accAddrs -> filterByAddrs (S.singleton addr)
             | otherwise                -> throw errorBadAddress
 
-    first filterFn <$> getFullWalletHistory cWalId
+    !res <- first filterFn <$> getFullWalletHistory cWalId
```
Comment (dcoutts) [inline wallet/src/Pos/Wallet/Web/Methods/History.hs]: This doesn't do what you think it does. This forces a pair but not its components! See comment above about where to force it.

Comment (adinapoli-iohk) [reply wallet/src/Pos/Wallet/Web/Methods/History.hs]: Oh dear, I haven't realised this code was using `first` from `Control.Arrow` and yielding a tuple. Of course, for a tuple forcing into WHNF won't do very much of a difference. Let me handle this.

**Comment by adinapoli-iohk** [comment]
@dcoutts Should be better-looking now.

**Pull request merged**